src/main/src/main/java/org/geoserver/security/password/ConfigurationPasswordEncryptionHelper.java

```diff
@@ -170,10 +173,9 @@ public String encode(String value) {
 
     /** Decrypts previously encrypted store connection parameters. */
     public void decode(StoreInfo info) {
-        List<GeoServerPasswordEncoder> encoders = securityManager.loadPasswordEncoders(null, true, null);
-
         Set<String> encryptedFields = getEncryptedFields(info);
-        if (info.getConnectionParameters() != null) {
+        if (!encryptedFields.isEmpty() && info.getConnectionParameters() != null) {
```

Comment:
`securityManager.loadPasswordEncoders()` comes with an overhead and is not necessary to call it if there are no encryptable fields to check